src/ORM/AssociationCollection.php

```diff
@@ -259,7 +259,14 @@ protected function _save($association, $entity, $nested, $options)
     public function cascadeDelete(Entity $entity, array $options)
     {
         foreach ($this->_items as $assoc) {
-            $assoc->cascadeDelete($entity, $options);
+            if ($assoc->cascadeCallbacks()) {
+                $assoc->cascadeDelete($entity, $options);
+            }
+        }
+        foreach ($this->_items as $assoc) {
```

Comment:
Looping twice over the full set of associations seems less than ideal. Could we loop and filter, so the second loop is shorter?